src/sentry/monitors/validators.py

```diff
@@ -527,3 +536,137 @@ def validate_ids(self, value):
+        return CronMonitorDataSourceHandler
+
+    def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
+        if not attrs.get("name") and not attrs.get("slug"):
+            raise serializers.ValidationError("Either name or slug must be provided")
+
+        if not attrs.get("name") and attrs.get("slug"):
+            attrs["name"] = attrs["slug"]
+
+        if attrs.get("name") and not attrs.get("slug") and not self.instance:
+            attrs["slug"] = slugify_monitor_slug(attrs["name"])
+
+        monitor_data = attrs.copy()
+
+        if "is_muted" in monitor_data:
+            monitor_data["isMuted"] = monitor_data.pop("is_muted")
+
+        monitor_data["project"] = self.context["project"].slug
+
+        monitor_instance = None
+        if self.instance:
+            monitor_instance = self.instance
+
+        monitor_validator = MonitorValidator(
+            data=monitor_data,
+            context=self.context,
+            instance=monitor_instance,
+            partial=self.partial,
+        )
+
+        if not monitor_validator.is_valid():
+            raise serializers.ValidationError(monitor_validator.errors)
+
+        attrs["_monitor_validator"] = monitor_validator
+
+        validated = monitor_validator.validated_data
+        if "name" in validated:
+            attrs["name"] = validated["name"]
+        if "slug" in validated:
+            attrs["slug"] = validated["slug"]
+        if "config" in validated:
+            attrs["config"] = validated["config"]
+        if "status" in validated:
+            attrs["status"] = validated["status"]
+        if "owner" in validated:
+            attrs["owner"] = validated["owner"]
+        if "is_muted" in validated:
+            attrs["is_muted"] = validated["is_muted"]
+
+        return super().validate(attrs)
+
+    def create_source(self, validated_data: dict[str, Any]) -> Monitor:
+        """Create the Monitor using MonitorValidator."""
+        monitor_validator = validated_data.pop("_monitor_validator")
+        with in_test_hide_transaction_boundary():
+            return monitor_validator.create(monitor_validator.validated_data)
+
+    def update(self, instance: Monitor, validated_data: dict[str, Any]) -> Monitor:
+        monitor_validator = validated_data.pop("_monitor_validator")
+        with in_test_hide_transaction_boundary():
+            return monitor_validator.update(instance, monitor_validator.validated_data)
+
+
+class MonitorIncidentDetectorValidator(BaseDetectorTypeValidator):
+    """
+    Validator for monitor incident detection configuration.
+
+    This is a lightweight validator that delegates Monitor creation/update to the
+    data_source field (MonitorDataSourceValidator).
+    """
+
+    data_source = MonitorDataSourceValidator(required=True)
+
+    def update(self, instance: Detector, validated_data: dict[str, Any]) -> Detector:
+        super().update(instance, validated_data)
+
+        if "data_source" in validated_data:
```

Comment:
i've been trying to think of ways to simplify this a bit for children of `BaseDetectorTypeValidator` -- any thoughts since you just wrote this? i feel like we could at _least_ provide a simplified hook for updating these. it also gets a lot crazier when there are multiple data sources.

Reply:
Yeah, I think we can have a hook here to handle these updates automatically rather than us having to add in our own handling logic each time. Not totally sure what that'd look like yet, but we can discuss